Templates/Swift/Sources/API.swift

```diff
@@ -14,8 +14,16 @@ public struct {{ options.name }} {
     public static var safeArrayDecoding = {% if options.safeArrayDecoding %}true{% else %}false{% endif %}
 
     /// Used to encode Dates when uses as string params
-    public static let dateEncodingFormatter = DateFormatter(formatString: "yyyy-MM-dd'T'HH:mm:ssZZZZZ")
-
+    public static var dateEncodingFormatter = DateFormatter(formatString: "yyyy-MM-dd'T'HH:mm:ssZZZZZ")
+    
+    /// Default JSONEncoder used to enconde each API request
+    public static var defaultJSONEncoder: JSONEncoder {
+        let jsonEncoder = JSONEncoder()
+        jsonEncoder.dateEncodingStrategy = .formatted({{ options.name }}.dateEncodingFormatter)
```

Comment:
Hi @jrcmramos, sorry for the delay on this, and thanks for updating! I'm not quite convinced that using a getter for the defaultEncoder is the best way to go. For this to be truly customizable it should be just a static variable, as people may want to change other things about the encoder. The instance can default to using the existing `dateEncodingFormatter`. 

```swift
public static var defaultJSONEncoder: JSONEncoder = {
    let jsonEncoder = JSONEncoder()
    jsonEncoder.dateEncodingStrategy = .formatted({{ options.name }}.dateEncodingFormatter)
    return jsonEncoder
}()
```

Reply:
Thanks for you feedback @yonaskolb! In fact it is more flexible your suggestion, I my reasoning was to always have the same strategy for both encoding and decoding and in case of an update of the decoder strategy, this one would also be affected, since it is always computed